### PR TITLE
feat: add signal_params for signal-processing constants

### LINKED ISSUE #156: [Feature] Add more customisability to method params
At the moment everything is quite rigid to true and tested settings that work well for daily time series.

Introduce parameters to 'pt.detect_trends(method_params)' to change constants that are currently hard set under the hood.  

These will have direct impact on logic in 'process_signals' and 'post_processing'

Must be handled in a certain way, where if some constants should depend on others then they should - to avoid breaking reliability.

This will help the user in cases like:
- they want to model trends on weekly aggregate points but don't want seasonality of 7 (like in daily) but 4 (for monthly)
- similarly they want to model monthly seasonality but with daily aggregate points, setting the trends to reasonably try account for seasonality of 28 days (for monthly)
- user has a particularly flat signal, that they want the method to be more sensitive on small motions of trend 
- user has a highly variable trend, where they want method to be less sensitive to small motions of trend.

Pytrendy is hard set to maximise detecting trends that could most generally be accepted as valid true positives. 

This update could take it a step further to allow flexibility for the users personal preferences - with reason.

### PR DESCRIPTION
**Pull request by RussellSB**
## Summary

Adds a dedicated **`signal_params`** surface for the signal-processing constants, independent from `method_params`. This branch was **restarted from scratch** per maintainer direction — the previous `method_params`-passthrough work was discarded with a sanctioned reset.

`method_params` keeps its three keys (`abrupt_padding`, `gradual_padding`, `avoid_noise`); the windows, thresholds and grouping/length constants move into `signal_params`, which is threaded through the pipeline. Refs #156.

## Design

**`detect_trends` owns all defaulting.** It is the single public entry point, so it is the one place defaults and derivations live: it calls the public `prep_signal_params.prep_signal_params()` (`pytrendy/io/prep_signal_params.py`), which merges `_SIGNAL_PARAMS_DEFAULTS` with the caller's overrides, then derives `window_flat` / `window_noise` from `window_smooth` when not supplied. Every stage receives an always-populated `signal_params` dict and reads it **directly** (`signal_params['window_smooth']`) — no `.get()` fallbacks, no re-derivation, no redeclared constants.

## Changes

- **New `signal_params` keyword** on `detect_trends(...)` — the user-facing override surface for signal-processing constants. Defaults resolve in one place (`prep_signal_params.prep_signal_params()` in `pytrendy/io/prep_signal_params.py`): `window_smooth 15` · `smooth_factor 0.5` · `noise_factor 0.5` · `grouping_distance 7` · `min_trend_length 3` · `min_flat_noise_length 1` · `threshold_noise 2.5` · `threshold_smooth 0.001` · `threshold_flat 0.835`; `window_flat`/`window_noise` derive from `window_smooth` via `smooth_factor`/`noise_factor` (the fractions of the smoothing window they span, 0.5 by default) unless explicitly overridden.
- **Threaded to every stage** — each takes a hard-typed `signal_params: dict` and reads values directly: `process_signals` (windows, thresholds, noise-grouping distance), `get_segments` (min lengths), `refine_segments` → `group_segments` (grouping distance, all four call sites) and `clean_artifacts` (grouping distance, `threshold_noise`). The `GROUPING_DISTANCE` constant and every stage-level default are removed.
- **Docs** — docstring key/default/effect table (window sizes in points, min lengths and grouping distance in steps, thresholds in dB/fraction), independence from `method_params`, and a non-daily cadence note (see #303/#309). Index staging renamed: `io/prepare_index.py` → `io/prep_index.py` (sibling helpers keep their names).
- Unknown keys are accepted and forwarded silently (no validation in v1).

## Verification

- `python -m pytest` (matplotlib 3.10.8): **175 passed**.
- New `tests/test_signal_params.py`: defaults identical to omission; `window_flat`/`window_noise` derive from `window_smooth` (via `smooth_factor`/`noise_factor`, explicit overrides win); each of `window_smooth`, `grouping_distance` (noise-grouping path), `min_trend_length` and `threshold_noise` overrides changes detection; unknown keys accepted silently.
- Coverage: ~99.9% total (single pre-existing `artifact_cleanup.py` edge-case miss, untouched by this change); `detect_trends.py` and every touched file 100%.
- Daily behaviour is byte-identical at defaults; no plot baselines touched.

## For review at the end

- **Naming / release impact:** `prepare_index` → `prep_index` renames a module that is rendered in the auto-generated API reference (mkdocs `api-autonav`), plus `_resolve_signal_params` → public `prep_signal_params` in `io/`. Both are internal-only (not exported from `pytrendy`), but flagging whether this warrants a breaking bump in the 2.0.0 stream.
- **Validation:** add did-you-mean validation for `signal_params` keys later (v1 accepts unknown keys silently)?
- **Coordination with #309:** #309 rebases onto this and migrates its window overrides to `signal_params`; the top-level defaulting block here is the pattern it builds on.

### CODE REVIEW

**Comment by RussellSB** [comment]
Redirecting: the scope I implemented (method_params extensibility) was a misreading of this issue. The intent was a separate `signal_params` surface for the signal-processing constants (WINDOW, NEIGHBOURING/GROUPING DISTANCE, thresholds) so they can be tuned independently of `method_params`. A fresh PR with that scope is being opened; this one is closed unmerged.

**Comment by RussellSB** [comment]
Restarting from scratch per maintainer clarification: this PR will be redone as a separate `signal_params` surface for the signal-processing constants (WINDOW sizes, NEIGHBOURING/GROUPING DISTANCE, thresholds, min lengths) — independent of `method_params`. The old method_params branch work is being reset; new commits incoming.

File: pytrendy/post_processing/segments_refine/artifact_cleanup.py [hunk 1]
```diff
@@ -182,7 +182,7 @@ def has_partial_overlap_prev(segment: dict, segment_prev: dict) -> bool:
 
     # Pass 3: Cleans partial overlaps with noise. Don't filter out completely when partial, adjust outside noise
     # Only runs post-processing cleanup logic when avoid_noise is True, enabled by default.
-    if method_params['avoid_noise']:
+    if method_params.get('avoid_noise', True):
```
Comment (RussellSB) [inline pytrendy/post_processing/segments_refine/artifact_cleanup.py]: Remove this off-topic git diff

File: pytrendy/post_processing/segments_refine/artifact_cleanup.py [hunk 2]
```diff
@@ -313,7 +313,7 @@ def has_partial_overlap_prev(segment: dict, segment_prev: dict) -> bool:
                 trend_too_flat = not min_in_last_section
 
         # Reclassify as noise if either edge cases met
-        if method_params['avoid_noise'] and \
+        if method_params.get('avoid_noise', True) and \
```
Comment (RussellSB) [inline pytrendy/post_processing/segments_refine/artifact_cleanup.py]: Remove this off-topic git diff

File: pytrendy/post_processing/segments_refine/segment_grouping.py [hunk 3]
```diff
@@ -8,20 +8,21 @@
 GROUPING_DISTANCE = 7  # Distance for grouping segments of same type in group_segments
```
Comment (RussellSB) [inline pytrendy/post_processing/segments_refine/segment_grouping.py]: Can clean this out since now deprecated in favour of passed parameter instead of constant

File: pytrendy/post_processing/segments_refine/__init__.py [hunk 4]
```diff
@@ -8,11 +8,11 @@
 from .trend_classify import classify_trends
 from .gradual_expand_contract import expand_contract_segments
 from .abrupt_shaving import shave_abrupt_trends
-from .segment_grouping import group_segments
+from .segment_grouping import group_segments, GROUPING_DISTANCE
```
Comment (RussellSB) [inline pytrendy/post_processing/segments_refine/__init__.py]: shouldnt GROUPING_DISTANCE be removed?

File: pytrendy/post_processing/segments_refine/__init__.py [hunk 5]
```diff
@@ -31,33 +31,39 @@ def refine_segments(df: pd.DataFrame, value_col: str, segments: list[dict], meth
...
+        signal_params (dict, optional): Optional signal-processing constants. Supported keys:
+
+            - **grouping_distance** (`int`): Maximum gap, in index steps, for grouping. Defaults to `7`.
 
     Returns:
         list: Final refined segment list.
     """
 
+    signal_params = signal_params or {}
+    grouping_distance = signal_params.get('grouping_distance', GROUPING_DISTANCE)
```
Comment (RussellSB) [inline pytrendy/post_processing/segments_refine/__init__.py]: no need to muddy it up, just assume that it is always passed from the top level.
If ommited just declare it as 7 from the top level, it should never reach this edge case since detect_trends() is the main accessible point

File: pytrendy/post_processing/segments_get.py [hunk 6]
```diff
@@ -20,12 +20,17 @@ def get_segments(df: pd.DataFrame) -> list[dict]:
 
     Only segments meeting the following minimum lengths are retained:
 
-    - Up/Down: ≥ 3 days
-    - Flat/Noise: ≥ 1 days
+    - Up/Down: ≥ 3 points (``min_trend_length``)
```
Comment (RussellSB) [inline pytrendy/post_processing/segments_get.py]: call it steps

File: pytrendy/post_processing/segments_get.py [hunk 7]
```diff
@@ -45,6 +50,10 @@ def get_segments(df: pd.DataFrame) -> list[dict]:
         , -3: 'Noise'
     }
 
+    signal_params = signal_params or {}
```
Comment (RussellSB) [inline pytrendy/post_processing/segments_get.py]: like for grouping distance, do not set any defaults messily in the code kind of spaghetti like. Let the top level do the heavy lifting in defaulting params that arent declared from signal_params

File: pytrendy/detect_trends.py [hunk 8]
```diff
@@ -11,11 +11,27 @@
 from .io import prepare_index
 
 
+# Signal-processing constants. Windows and minimum lengths are in points;
+# `grouping_distance` is in index steps; thresholds are in dB / fraction.
```
Comment (RussellSB) [inline pytrendy/detect_trends.py]: move the comments to be line by line,
e.g. grouping distance comment on line 20, after the code,
threshold_smooth comment on line 24 after the code etc...

File: pytrendy/process_signals.py [hunk 9]
```diff
@@ -7,7 +7,7 @@
 from scipy.stats import iqr
 from .post_processing.segments_refine.segment_grouping import GROUPING_DISTANCE
 
-def process_signals(df: pd.DataFrame, value_col: str, method_params: dict, debug: bool=False) -> pd.DataFrame:
+def process_signals(df: pd.DataFrame, value_col: str, method_params: dict, signal_params: dict|None=None, debug: bool=False) -> pd.DataFrame:
```
Comment (RussellSB) [inline pytrendy/process_signals.py]: just make it :dict to follow convention. It is fine. Can refactor more widely later. Check other files for this. Hard type jsut dict, dont worry too much about None

File: pytrendy/process_signals.py [hunk 10]
```diff
@@ -46,13 +56,17 @@ def process_signals(df: pd.DataFrame, value_col: str, method_params: dict, debug
...
-    THRESHOLD_NOISE = 2.5 # Sensitivity to detecting noise (recommended 0-10)
-    THRESHOLD_SMOOTH = 0.001 # Sensitivity to detecting trends as fraction of iqr
-    THRESHOLD_FLAT = 0.835 # Sensitivity to detecting flats as a fraction of min std (non-zero)
+    WINDOW_SMOOTH = signal_params.get('window_smooth', 15)
+    WINDOW_FLAT = signal_params.get('window_flat') or int(WINDOW_SMOOTH*0.5)
+    WINDOW_NOISE = signal_params.get('window_noise') or int(WINDOW_SMOOTH*0.5)
+
+    THRESHOLD_NOISE = signal_params.get('threshold_noise', 2.5) # Sensitivity to detecting noise (recommended 0-10)
+    THRESHOLD_SMOOTH = signal_params.get('threshold_smooth', 0.001) # Sensitivity to detecting trends as fraction of iqr
+    THRESHOLD_FLAT = signal_params.get('threshold_flat', 0.835) # Sensitivity to detecting flats as a fraction of min std (non-zero)
```
Comment (RussellSB) [inline pytrendy/process_signals.py]: again let the top level do the heavy lifting !!!! and dont redeclare the vars and defaults.
Maintenance overhead. Simply refer to top level signal_parms variables like signal_params['threshold_flat'] . When you need it. Safe to assume it will always be populated as handled well from top level

File: pytrendy/post_processing/segments_refine/__init__.py [hunk 11]
```diff
@@ -8,11 +8,11 @@
 from .trend_classify import classify_trends
 from .gradual_expand_contract import expand_contract_segments
 from .abrupt_shaving import shave_abrupt_trends
-from .segment_grouping import group_segments
+from .segment_grouping import group_segments, GROUPING_DISTANCE
 from .artifact_cleanup import clean_artifacts, fill_in_flats
 
 
-def refine_segments(df: pd.DataFrame, value_col: str, segments: list[dict], method_params: dict) -> list[dict]:
+def refine_segments(df: pd.DataFrame, value_col: str, segments: list[dict], method_params: dict, signal_params: dict|None=None) -> list[dict]:
```
Comment (RussellSB) [inline pytrendy/post_processing/segments_refine/__init__.py]: remove |None=None

File: pytrendy/post_processing/segments_refine/artifact_cleanup.py (hunk 1 above)
Comment (RussellSB) [reply pytrendy/post_processing/segments_refine/artifact_cleanup.py]: [opencode-go/glm-5.3-flash]: All addressed in `99b829f2` (head `d388ab94`). Per your notes:

- **R185/R316** (`artifact_cleanup.py` off-topic hunks): the only off-topic change was the `avoid_noise` hygiene rider; reverted to `method_params['avoid_noise']`. The remaining hunks are the `signal_params` threading you asked for.
- **R8/R11** (`GROUPING_DISTANCE`): constant deleted from `segment_grouping.py`, and its imports removed from `segments_refine/__init__.py` and `artifact_cleanup.py`. The default `7` is now declared once, in `_SIGNAL_PARAMS_DEFAULTS`.
- **R43** (noise-grouping path): reads `signal_params['grouping_distance']` directly — no defensive fallback.
- **R53/R67** (messy inline defaulting): the whole local `WINDOW_*`/`THRESHOLD_*` block is gone from `process_signals` (including the `int(WINDOW_SMOOTH*0.5)` re-derivation); `window_flat`/`window_noise` are derived once in `_resolve_signal_params()` at the top level.
- **R15** (comment layout): per-entry inline comments in `_SIGNAL_PARAMS_DEFAULTS` — grouping-distance comment on its own line, threshold comments on theirs.
- **R10/R15** (typing): `signal_params: dict` (no `|None`) on `process_signals`, `get_segments`, `refine_segments` and `clean_artifacts`; `detect_trends` keeps `dict | None = None` as the user-facing entry point.
- **R23** (wording): min-length docs and variable names now say "steps" throughout.

Added a test covering the `window_flat`/`window_noise` derivation plus explicit overrides. Suite: 174 passed; daily behaviour byte-identical.

File: tests/test_signal_params.py [hunk 12]
```diff
@@ -0,0 +1,87 @@
...
+        """grouping_distance controls the noise-segment grouping in process_signals."""
+        df = pt.load_data('series_synthetic')
+        df_int, _, _, _ = prepare_index.prepare_index(df.copy(), 'date', 'gradual-noisy-20')
+
+        default = process_signals(df_int.copy(), 'gradual-noisy-20', {'avoid_noise': True}, _resolve_signal_params())
+        overridden = process_signals(
+            df_int.copy(), 'gradual-noisy-20', {'avoid_noise': True}, _resolve_signal_params({'grouping_distance': 0})
+        )
+        # Disabling grouping must keep (at least as many) separate noise regions.
+        assert int(overridden['noise_flag'].sum()) > int(default['noise_flag'].sum())
```
Comment (RussellSB) [inline tests/test_signal_params.py]: make this >=

File: pytrendy/detect_trends.py [hunk 13]
```diff
@@ -11,11 +11,45 @@
...
+    'grouping_distance': 7,     # Maximum gap, in index steps, for grouping nearby segments.
+    'min_trend_length': 3,      # Minimum length, in steps, for an Up/Down segment to be retained.
+    'min_flat_noise_length': 1, # Minimum length, in steps, for a Flat/Noise segment to be retained.
+    'threshold_noise': 2.5,     # SNR threshold (dB) below which a region is classified as noise.
+    'threshold_smooth': 0.001,  # Derivative threshold as a fraction of the signal IQR, below which motion counts as flat.
+    'threshold_flat': 0.835,    # Flat sensitivity as a fraction of the minimum non-zero rolling std.
+}
+
+
+def _resolve_signal_params(signal_params: dict|None=None) -> dict:
```
Comment (RussellSB) [inline pytrendy/detect_trends.py]: dont make this private, and move this to io call it prep_signal_params. and refector prepare_index to prep_index

File: pytrendy/post_processing/segments_refine/artifact_cleanup.py (hunk 1 above)
Comment (RussellSB) [reply pytrendy/post_processing/segments_refine/artifact_cleanup.py]: [opencode-go/deepseek-v4.1-flash]: The `avoid_noise` hygiene rider was the only off-topic hunk — reverted to `method_params['avoid_noise']` in `99b829f2`.

File: pytrendy/post_processing/segments_refine/artifact_cleanup.py (hunk 2 above)
Comment (RussellSB) [reply pytrendy/post_processing/segments_refine/artifact_cleanup.py]: [opencode-go/deepseek-v4.1-flash]: The `avoid_noise` hygiene rider was the only off-topic hunk — reverted to `method_params['avoid_noise']` in `99b829f2`.

File: pytrendy/post_processing/segments_refine/segment_grouping.py (hunk 3 above)
Comment (RussellSB) [reply pytrendy/post_processing/segments_refine/segment_grouping.py]: [opencode-go/deepseek-v4.1-flash]: `GROUPING_DISTANCE` constant deleted from `segment_grouping.py` with its imports; default 7 now declared once in `_SIGNAL_PARAMS_DEFAULTS` in `pytrendy/io/prep_signal_params.py`.

File: pytrendy/post_processing/segments_refine/__init__.py (hunk 4 above)
Comment (RussellSB) [reply pytrendy/post_processing/segments_refine/__init__.py]: [opencode-go/deepseek-v4.1-flash]: `GROUPING_DISTANCE` constant deleted from `segment_grouping.py` with its imports; default 7 now declared once in `_SIGNAL_PARAMS_DEFAULTS` in `pytrendy/io/prep_signal_params.py`.

File: pytrendy/post_processing/segments_refine/__init__.py (hunk 5 above)
Comment (RussellSB) [reply pytrendy/post_processing/segments_refine/__init__.py]: [opencode-go/deepseek-v4.1-flash]: Now reads `signal_params['grouping_distance']` directly — no fallback (`99b829f2`).

File: pytrendy/post_processing/segments_get.py (hunk 6 above)
Comment (RussellSB) [reply pytrendy/post_processing/segments_get.py]: [opencode-go/deepseek-v4.1-flash]: Min-length wording is now 'steps' throughout (`d388ab94`).

File: pytrendy/post_processing/segments_get.py (hunk 7 above)
Comment (RussellSB) [reply pytrendy/post_processing/segments_get.py]: [opencode-go/deepseek-v4.1-flash]: The whole local `WINDOW_*`/`THRESHOLD_*` defaulting block is gone; windows/thresholds read `signal_params['…']` directly, defaults/derivations live once at the top level.

File: pytrendy/detect_trends.py (hunk 8 above)
Comment (RussellSB) [reply pytrendy/detect_trends.py]: [opencode-go/deepseek-v4.1-flash]: Comments moved line-by-line onto their code lines in `e48509dc`.

File: pytrendy/process_signals.py (hunk 9 above)
Comment (RussellSB) [reply pytrendy/process_signals.py]: [opencode-go/deepseek-v4.1-flash]: `signal_params: dict` on all stage signatures; `detect_trends` keeps `dict | None = None` as the user-facing entry point (`99b829f2`).

File: pytrendy/process_signals.py (hunk 10 above)
Comment (RussellSB) [reply pytrendy/process_signals.py]: [opencode-go/deepseek-v4.1-flash]: The whole local `WINDOW_*`/`THRESHOLD_*` defaulting block is gone; windows/thresholds read `signal_params['…']` directly, defaults/derivations live once at the top level.

File: pytrendy/post_processing/segments_refine/__init__.py (hunk 11 above)
Comment (RussellSB) [reply pytrendy/post_processing/segments_refine/__init__.py]: [opencode-go/deepseek-v4.1-flash]: `signal_params: dict` on all stage signatures; `detect_trends` keeps `dict | None = None` as the user-facing entry point (`99b829f2`).

File: tests/test_signal_params.py (hunk 12 above)
Comment (RussellSB) [reply tests/test_signal_params.py]: [opencode-go/deepseek-v4.1-flash]: Changed to `>=` in `48c90258` — the boundary case is now asserted.

File: pytrendy/detect_trends.py (hunk 13 above)
Comment (RussellSB) [reply pytrendy/detect_trends.py]: [opencode-go/deepseek-v4.1-flash]: `_resolve_signal_params` is now public `prep_signal_params` in `pytrendy/io/prep_signal_params.py`, and `prepare_index` → `prep_index` (module + main function) repo-wide — commits `535635e7`/`125bb7ac`. Flagged in the body's For review whether the docs-rendered module rename warrants a breaking bump.

File: pytrendy/post_processing/segments_get.py (hunk 6 above)
Comment (RussellSB) [reply pytrendy/post_processing/segments_get.py]: you called it point actually call it steps

File: pytrendy/post_processing/segments_get.py (hunk 7 above)
Comment (RussellSB) [reply pytrendy/post_processing/segments_get.py]: still just rely on signal params handling .get default conditions top level. remove line 53, and dont add .get defaults for lines 54-55. 

Find similar patterns else where in the codebase and correct. All defaults for signal params should be handled from the top level and top level only.

File: pytrendy/detect_trends.py (hunk 8 above)
Comment (RussellSB) [reply pytrendy/detect_trends.py]: still not moved ! E.g.

_SIGNAL_PARAMS_DEFAULTS = {
    'window_smooth': 15, # `window_smooth` unless explicitly overridden.
    'grouping_distance': 7, # `grouping_distance` is in index steps; thresholds are in dB / fraction.
    .... etc
}

window flat and window noise can be the exception to remain on top

File: pytrendy/process_signals.py (hunk 9 above)
Comment (RussellSB) [reply pytrendy/process_signals.py]: No make this `signal_params: dict` , no `|None=None` here

File: pytrendy/process_signals.py (hunk 10 above)
Comment (RussellSB) [reply pytrendy/process_signals.py]: Again let the defaults be handled top level from io prep_signal_params!

File: tests/test_signal_params.py (hunk 12 above)
Comment (RussellSB) [reply tests/test_signal_params.py]: cant see?

File: pytrendy/post_processing/segments_get.py (hunk 6 above)
Comment (RussellSB) [reply pytrendy/post_processing/segments_get.py]: [opencode-go/glm-5.3-flash]: This was already fixed by the time of your comment — it was anchored to `43b9e3d1`. Min-length wording is `steps` at head (`get_segments` docs, `detect_trends`, `prep_signal_params`); window sizes stay `points` per the accepted design distinction.

File: pytrendy/post_processing/segments_get.py (hunk 7 above)
Comment (RussellSB) [reply pytrendy/post_processing/segments_get.py]: [opencode-go/glm-5.3-flash]: Also already fixed at head — your snippet is the stale `43b9e3d1` thread state. Both min-length keys are read directly (`signal_params['min_trend_length']` / `['min_flat_noise_length']`). The full sweep found one remaining hardcoded stage default — `threshold_noise = 2.5` in `clean_artifacts` — now `signal_params['threshold_noise']` (`2f425bee`). No `.get`-with-default signal_params reads remain in stage code; all defaulting lives in `prep_signal_params` (io).

File: pytrendy/detect_trends.py (hunk 8 above)
Comment (RussellSB) [reply pytrendy/detect_trends.py]: [opencode-go/glm-5.3-flash]: The per-entry inline comments were in place, but a duplicated block comment sat above the dict — dropped in `6cca550f` so the block matches your sketch exactly (defaults now live in `pytrendy/io/prep_signal_params.py`).

File: pytrendy/io/prep_signal_params.py [hunk 14]
```diff
@@ -0,0 +1,42 @@
...
+    this returns a fully-populated dict.
+
+    Args:
+        signal_params (dict, optional): User overrides. Unknown keys are forwarded silently.
+
+    Returns:
+        dict: Fully-populated signal_params with every key the stages read.
+    """
+    resolved = {**_SIGNAL_PARAMS_DEFAULTS, **(signal_params or {})}
+    resolved.setdefault('window_flat', int(resolved['window_smooth'] * 0.5))
```
Comment (RussellSB) [inline pytrendy/io/prep_signal_params.py]: I think it can be worth setting these thresholds to be passed through to as params...
e.g. smooth_factor, noise_factor, defaulted at 0.5. And with comment to explain what it is how nudging up/down changes behaviour and briefly how/why it applies to window_smooth to keep things consistent

File: pytrendy/detect_trends.py [hunk 15]
```diff
@@ -51,6 +52,22 @@ def detect_trends(df: pd.DataFrame,
...
+            - **grouping_distance** (`int`): Maximum gap, in steps, for grouping nearby segments. Defaults to `7`.
+            - **min_trend_length** (`int`): Minimum length, in steps, for an Up/Down segment to be retained. Defaults to `3`.
+            - **min_flat_noise_length** (`int`): Minimum length, in steps, for a Flat/Noise segment to be retained. Defaults to `1`.
+            - **threshold_noise** (`float`): SNR threshold (dB) below which a region is classified as noise. Defaults to `2.5`.
+            - **threshold_smooth** (`float`): Derivative threshold, as a fraction of the signal IQR, below which motion counts as flat. Defaults to `0.001`.
+            - **threshold_flat** (`float`): Flat sensitivity, as a fraction of the minimum non-zero rolling std. Defaults to `0.835`.
+
+            This surface is independent from `method_params`, which controls the padding and noise heuristics instead.
+            Window sizes are in points and minimum lengths in steps; at non-daily spacing a given count spans a
+            different real-time duration (see #303/#309 for the cadence-aware view).
```
Comment (RussellSB) [inline pytrendy/detect_trends.py]: remove this part of the comment: `(see #303/#309 for the cadence-aware view).` since API docs facing

**Comment by RussellSB** [comment]
[opencode-go/glm-5.3-flash]: Done — the internal cross-reference is gone: the `signal_params` docstring note now reads "…at non-daily spacing a given count spans a different real-time duration." with no PR references (API-docs-facing). Committed and pushed as `735985f4` (`docs: drop internal cross-reference from docstring`), suite 175 passed, daily behaviour unchanged.
